src/Zicht/Bundle/SolrBundle/Solr/Client.php

```diff
@@ -83,19 +87,29 @@ protected function doRequest(QueryBuilder\RequestBuilderInterface $handler)
             }
 
             $this->logs[] = ['response' => $response, 'requestUri' => $request->getUri()];
-        } catch (BadResponseException $e) {
-            $this->lastResponse = $e->getResponse();
-            if ($e->getRequest()->getBody()) {
-                $e->getRequest()->getBody()->seek(0);
+        } catch (BadResponseException $be) {
+            $this->lastResponse = $be->getResponse();
+            if ($be->getRequest()->getBody()) {
+                $be->getRequest()->getBody()->seek(0);
+            }
+            $content = $be->getResponse()->getBody()->getContents();
+            $contentType = $this->lastResponse->getHeaderLine('Content-Type');
+            $errorMsg = $be->getMessage();
+            if (preg_match('!^application/json!', $contentType) || preg_match('!^text/plain!', $contentType)) {
+                // possibly content is a json-string containing a SolrException.
+                $solrException = \GuzzleHttp\json_decode($content);
```

Comment:
The `\GuzzleHttp\json_decode` throws a `\InvalidArgumentException` so i think you should do a catch/try around the  `json_decode` because this could lead to a exception that has noting to do with the request.

Reply:
Done.